test/record_runner.zig

```diff
@@ -86,9 +85,6 @@ const ExpectedFailure = struct {
 const builtin = @import("builtin");
 
 pub fn main() !void {
-    if (builtin.os.tag == .windows and (builtin.mode == .ReleaseFast or builtin.mode == .ReleaseSmall)) {
-        return; // Building the test runner in release modes crashes the Zig compiler.
```

Comment:
Was there a Zig issue related to these checks?

Reply:
https://github.com/ziglang/zig/issues/17332

Reply:
Maybe I should not have removed the windows one - I can't check that easily.

Reply:
FWIW it builds fine with `zig build test -Dtarget=x86_64-windows -Doptimize=ReleaseFast` (and `ReleaseSmall`); I just don't have a way to run the tests currently.

Reply:
```console
❯ uname -a
MINGW64_NT-10.0-26120 alexrp 3.5.3.x86_64 2024-06-03 06:22 UTC x86_64 Msys
❯ zig version
0.13.0-dev.351+64ef45eb0
❯ git clone git@github.com:ehaas/arocc.git
Cloning into 'arocc'...
remote: Enumerating objects: 11851, done.
remote: Counting objects: 100% (3795/3795), done.
remote: Compressing objects: 100% (749/749), done.
remote: Total 11851 (delta 3203), reused 3397 (delta 3008), pack-reused 8056
Receiving objects: 100% (11851/11851), 4.09 MiB | 8.28 MiB/s, done.
Resolving deltas: 100% (8930/8930), done.
❯ cd arocc
❯ git checkout progress-updates
Switched to a new branch 'progress-updates'
branch 'progress-updates' set up to track 'origin/progress-updates' by rebasing.
❯ zig build test
max mem used = 10.80MiB
9236 passed; 117 skipped (174 invalid targets).
#pragma pack msvc: 1 test skipped
#pragma pack: 1 test skipped
attributes: 1 test skipped
constexpr: 1 test skipped
enum overflow: 2 tests skipped
hello world: 1 test skipped
incomplete types: 1 test skipped
initializers: 3 tests skipped
statement expressions: 1 test skipped
typeof: 1 test skipped
_Float16: 1 test skipped
310 passed; 14 skipped.
❯ zig build test -Doptimize=ReleaseFast
max mem used = 9.13MiB
9236 passed; 117 skipped (174 invalid targets).
#pragma pack msvc: 1 test skipped
#pragma pack: 1 test skipped
attributes: 1 test skipped
constexpr: 1 test skipped
enum overflow: 2 tests skipped
hello world: 1 test skipped
incomplete types: 1 test skipped
initializers: 3 tests skipped
statement expressions: 1 test skipped
typeof: 1 test skipped
_Float16: 1 test skipped
310 passed; 14 skipped.
❯ zig build test -Doptimize=ReleaseSmall
max mem used = 9.13MiB
9236 passed; 117 skipped (174 invalid targets).
#pragma pack msvc: 1 test skipped
#pragma pack: 1 test skipped
attributes: 1 test skipped
constexpr: 1 test skipped
enum overflow: 2 tests skipped
hello world: 1 test skipped
incomplete types: 1 test skipped
initializers: 3 tests skipped
statement expressions: 1 test skipped
typeof: 1 test skipped
_Float16: 1 test skipped
310 passed; 14 skipped.
```

Seems fine?